app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CommentsController < ApplicationController
+  before_action :set_post
+
+    def create
+     @comment = @post.comments.create(comment_params)
+    redirect_to @post
+  end
+  
+  def edit
+    @comment = Comment.find(params[:id])
+  end
+
+  def update
+    @comment = Comment.find(params[:id])
+    @comment.update(comment_params)
+    redirect_to @post
+  end
+
+  def destroy
+    @Comment = Comment.find(params[:id])
+    @comment.destroy
+    redirect_to @post
+  end
+
+  private
+  def comment_params
+    params.require(:comment).permit(:name, :content)
+  end
+  
+  private
```

Comment:
you don't have to write "Private " twice, you can do it just once and put necessary code blocks under that.